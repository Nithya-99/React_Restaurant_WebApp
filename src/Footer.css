.contact{
    color: #FFD700;
    text-decoration: none;
}
.page-footer{
    background-color: black;
}
.contact:hover{
    color:  white;
    text-decoration: none;
}
