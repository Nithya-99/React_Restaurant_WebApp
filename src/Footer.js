import React from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import './Footer.css'
class Footer extends React.Component {
    render() {
        return (
            <footer className="page-footer font-small fixed-bottom">
                <div className="text-center py-3 contact" ><b>Contact:</b>
                    <a data-testid="phn"> +91 9768304455</a> <br/>
                    <b>Address:</b>
                    <a data-testid="address"> 1st Floor, ATL Corporate Park, Saki Vihar Rd, Powai, Mumbai, Maharashtra 400072</a>
                </div>
            </footer>
        )
    }
}
export default Footer;