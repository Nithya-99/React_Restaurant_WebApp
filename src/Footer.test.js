import React from "react";
import { render, fireEvent, cleanup, wait } from "@testing-library/react";
import Footer from "./Footer";
import '@testing-library/jest-dom/extend-expect';

describe('Test Footer', ()=>{
    it('renders properly', ()=>{
        const {baseElement} = render(<Footer/>);
        expect(baseElement).toBeInTheDocument();
    });

    it('Check for contents in footer', ()=>{
        const {getByTestId} = render(<Footer/>);

        const phn = getByTestId('phn');
        expect(phn).toBeInTheDocument();

        const address = getByTestId('address');
        expect(address).toBeInTheDocument();
    })

    it('matches snapshot', ()=> {
        const {baseElement} = render(<Footer/>);
        expect(baseElement).toMatchSnapshot();
    });
});
  