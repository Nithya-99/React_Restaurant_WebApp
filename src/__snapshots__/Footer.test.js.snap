// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Footer matches snapshot 1`] = `
<body>
  <div>
    <footer
      class="page-footer font-small fixed-bottom"
    >
      <div
        class="text-center py-3 contact"
      >
        <b>
          Contact:
        </b>
        <a
          data-testid="phn"
        >
           +91 9768304455
        </a>
         
        <br />
        <b>
          Address:
        </b>
        <a
          data-testid="address"
        >
           1st Floor, ATL Corporate Park, Saki Vihar Rd, Powai, Mumbai, Maharashtra 400072
        </a>
      </div>
    </footer>
  </div>
</body>
`;
